src/classes/MergeService.cls

```diff
@@ -24,11 +26,21 @@ global with sharing class MergeService {
      */
     public MergeService(String sobjectName) {
         this.sobjectName = sobjectName;
+        this.sobjectNameOriginal = sobjectName;
+            
         this.concerns = [
             SELECT Id, FieldName__c, IsForceOverwrite__c
             FROM MergeConcern__mdt
             WHERE SobjectName__c = :sobjectName
         ];
+        // If it's person account switch the sobject name back to accounts
+        // by having them seperate it allows each to have seperate concerns saved in the custom metadata
+        // added by danhowellnz
+        if(sobjectName == 'PersonAccount'){
+            this.sobjectName = 'Account';
```

Comment:
Agreed. I think the whole service should use the Schema.SObjectType.Account.getName() on the first result being passed to the service.